store/src/java/com/zimbra/cs/html/DefangFilter.java

```diff
@@ -465,13 +466,30 @@ public void startDocument(XMLLocator locator, String encoding,
         DebugConfig.defangStyleUnwantedStrgPattern, Pattern.CASE_INSENSITIVE);
     
     private static String sanitizeStyleValue(String value) {
-        // remove comments
-        value = STYLE_UNWANTED_STRG_PATTERN.matcher(value).replaceAll("");
-        value = COMMENT.matcher(value).replaceAll("");
-        // strip off unwanted functions
-        value = STYLE_UNWANTED_FUNC.matcher(value).replaceAll("");
-       // strip off any @import
-        return STYLE_UNWANTED_IMPORT.matcher(value).replaceAll("");
+        if (value.length() < DebugConfig.defangStyleValueLimit) {
+            // remove comments
+            value = STYLE_UNWANTED_STRG_PATTERN.matcher(value).replaceAll("");
+            value = COMMENT.matcher(value).replaceAll("");
+            // strip off unwanted functions
+            Matcher matcher = STYLE_UNWANTED_FUNC.matcher(value);
+            StringBuffer stringBuffer = new StringBuffer();
+            while (matcher.find()) {
+                String match = matcher.group();
+                if (!match.startsWith("rgb") && !match.startsWith("media")
+                    && !match.startsWith("and")) {
+                    matcher.appendReplacement(stringBuffer, "");
+                }
+            }
+            matcher.appendTail(stringBuffer);
+            value = stringBuffer.toString();
+            // strip off any @import
+            ZimbraLog.mailbox.info("sanitizeStyleValue END");
+            value = STYLE_UNWANTED_IMPORT.matcher(value).replaceAll("");
+        } else {
+            ZimbraLog.mailbox.info("style value is too long:%d characters. Trimming it.", value.length());
+            value = "";
```

Comment:
Sneha you may want to return the original value, if users have emails with value for style larger than 10K but still valid they may complain